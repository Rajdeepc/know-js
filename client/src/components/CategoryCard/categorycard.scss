.category-card{
    .panel.panel-default {
        border-color: #5E5780;
        margin-bottom: 21px;
        background-color: #fff;
        border: 1px solid #5E5780;
        border-radius: 2px;
        position: relative;
        box-shadow: none;
        .panel-body {
            padding: 15px;
            .top-category--panel-content {
                display: flex;
                flex-direction: row;
                .top-category--category {
                    height: 32px;
                    line-height: 32px;
                    overflow: hidden;
                    text-overflow: ellipsis;
                    white-space: nowrap;
                    flex: 1;
                    min-width: 1px;
                    img{
                        width:30px
                    }
                }
            }
        }
    }
}