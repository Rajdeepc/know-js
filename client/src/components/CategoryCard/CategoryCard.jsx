import React from "react";
import { Image,Row,Col } from "react-bootstrap";
import './categorycard.scss';
import PropTypes from 'prop-types';


const CategoryCard = ({ item }) => {
  return (
    <div className="category-card">
      <div className="panel panel-default">
        <div className="panel-body">
          <a href="/courses/development/">
            <div className="top-category--panel-content">
              <Row>
                <Col>
                <div className="top-category--category">
                  <Image src={item.img} alt={item.core} />
                  </div>
                </Col>
                <Col>
                  <div className="top-category--category">{item.core}</div>
                </Col>
              </Row>
            </div>
          </a>
        </div>
      </div>
    </div>
  );
}

CategoryCard.propTypes = {
  img: PropTypes.string,
  core: PropTypes.string
}

CategoryCard.defaultProps = {
  img: '',
  core: ''
}

export default CategoryCard;
