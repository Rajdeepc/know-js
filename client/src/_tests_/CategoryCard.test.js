import React from 'react';
import { render, cleanup } from '@testing-library/react'
import CategoryCard from '../components/CategoryCard/CategoryCard';
import '@testing-library/jest-dom/extend-expect'

afterEach(cleanup);

const props = {
    item:{
    img:"",
    core: 'hello'
    }
}

describe('<Category component>', () => {
    test('should be defined', () => {
        expect(CategoryCard).toBeDefined();
    });
    test('renders Category Component', () => {
        const { container, rerender } = render(<CategoryCard {...props} />)
        rerender(<CategoryCard {...props} />)
        expect(container.firstChild).toMatchSnapshot()
      })
})
