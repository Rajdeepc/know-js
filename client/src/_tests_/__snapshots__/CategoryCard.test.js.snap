// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Category component> renders Category Component 1`] = `
<div
  class="category-card"
>
  <div
    class="panel panel-default"
  >
    <div
      class="panel-body"
    >
      <a
        href="/courses/development/"
      >
        <div
          class="top-category--panel-content"
        >
          <div
            class="row"
          >
            <div
              class="col"
            >
              <div
                class="top-category--category"
              >
                <img
                  alt="hello"
                  class=""
                  src=""
                />
              </div>
            </div>
            <div
              class="col"
            >
              <div
                class="top-category--category"
              >
                hello
              </div>
            </div>
          </div>
        </div>
      </a>
    </div>
  </div>
</div>
`;
